### PR TITLE
Fix custom rulesets not importing scores at all

### LINKED ISSUE #17350: Scores made with third party rulesets can't be saved
### Type

Game behaviour

### Bug description

Users are greeted with a "Score failed to import" notification after completing a play with a third party ruleset, and scores don't show in local leaderboards, and replays can't be viewed. 

This regression seems to be introduced in #17017, when the check in `LegacyScoreEncoder` have switched from checking `score.ScoreInfo.BeatmapInfo.Ruleset.OnlineID` which is always 0, to checking `score.ScoreInfo.Ruleset.OnlineID` which is always -1 for third party rulesets.

### Screenshots or videos

https://user-images.githubusercontent.com/12001167/159126084-eabe3e95-f459-4619-a656-92ef41b31176.mp4

### Version

2022.319.0

### Logs

[runtime.log](https://github.com/ppy/osu/files/8309611/runtime.log)
[updater.log](https://github.com/ppy/osu/files/8309612/updater.log)
[database.log](https://github.com/ppy/osu/files/8309613/database.log)
[network.log](https://github.com/ppy/osu/files/8309614/network.log)
[performance.log](https://github.com/ppy/osu/files/8309615/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/8309616/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/8309617/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/8309618/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/8309619/performance-update.log)

### PR DESCRIPTION
**Pull request by peppy**
Replaces the error with the ability to import, minus replays.

Closes https://github.com/ppy/osu/issues/17350 (arguably, but let's go with it for now).

### CODE REVIEW

**Comment by smoogipoo** [comment]
This replaces https://github.com/ppy/osu/pull/17787, correct?

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game/Tests/Visual/PlayerTestScene.cs [hunk 1]
```diff
@@ -22,6 +25,11 @@ public abstract class PlayerTestScene : RateAdjustedBeatmapTestScene
         /// </summary>
         protected virtual bool HasCustomSteps => false;
 
+        /// <summary>
+        /// WARNING: ONLY WORKS IF RUN HEADLESS because reasons.
+        /// </summary>
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Tests/Visual/PlayerTestScene.cs:30]: I can try to move it, but wedging in the correct place is frustrating as anything, and trying to find other similar usages is nigh impossible.

File: osu.Game.Tests/Visual/Gameplay/TestScenePlayerLocalScoreImport.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        {
+            Ruleset createCustomRuleset() => new OsuRuleset
+            {
+                RulesetInfo =
+                {
+                    Name = "custom",
+                    ShortName = "custom",
+                    OnlineID = -1
+                }
+            };
```
[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Tests/Visual/Gameplay/TestScenePlayerLocalScoreImport.cs:57]: Tempted to just drop the tests at this point.

[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Tests/Visual/Gameplay/TestScenePlayerLocalScoreImport.cs:57]: @frenzibyte for my own sanity, can you confirm that you meant "not enough to TEST importing scores"?

Safari 2022-07-08 at 09 42 29

[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game.Tests/Visual/Gameplay/TestScenePlayerLocalScoreImport.cs]: Presuming that's what you meant, this should now be fixed. Have tested that the new tests does fail against `master`.

**Review by frenzibyte — APPROVED** [review]

**Pull request merged**